GitHubProjectCards.ps1

```diff
@@ -604,6 +610,13 @@ filter Move-GitHubProjectCard
         $hashBody.add('column_id', $Column)
     }
 
+    if (-not $PSCmdlet.ShouldProcess($Card, 'Move GitHub Project Card'))
+    {
+        return
+    }
+
+    Write-InvocationLog
```

Comment:
A problem with this pattern (of putting `Write-InvocationLog` _after_ the `$PSCmdlet.ShouldProcess` is that error messages written to the log (like at line 588) won't have the invocation attribution to go along with it.

Reply:
But the invocation may never have happened if the user passed `whatif` to the function or selected `No` at confirm.

Reply:
But the user _did_ call it.  They just happened to call it with `-WhatIf`.  It still happened, and the invocation log will capture the fact that it was called with `-WhatIf`.  Some of these functions will print out errors as a result of processing the arguments before they ever call `Invoke-GHRestMethod`.  I contend that we want to see the invocation line before any such error messages in the logs in order to have context as to why those errors are happening.

**Scenario:  User runs `Move-GitHubProjectCard -Card -Top -Bottom -WhatIf`**
**With your current change**, there will be an error reported on console and in the log with "`You must use one (and only one) of the parameters Top, Bottom, or After.`"  This will be a disembodied error, because there will be nothing in the log to indicate _what_ caused that error to happen.

**If we keep the `Write-InvocationLog` to the top**, then it will capture the call as:
```
2020-07-15 10:35:34 : Howard : VERBOSE : [0.14.0] Executing: Move-GitHubProjectCard -Card -Top -Bottom -WhatIf
2020-07-15 10:35:34 : Howard : ERROR : You must use one (and only one) of the parameters Top, Bottom or After.
```
which is much more useful.  We see the error, but we also see the calling line to understand _why_ the error happened.

Users have the ability to filter out entries in the log file that contain `-WhatIf` if they don't find those relevant, but they have no way of getting context to disembodied errors if we postpone the invocation log entry until after user confirmation, unless we move the user confirmation to the _very top_ of the function, and we've previously agreed that the parameter processing _before_ the user confirmation has value.

**I do see your point however** for scenarios where `$ConfirmPreference -ne 'None'` and a user neither passed-in `-Force` nor `-Confirm:$false`.  In those scenarios, I believe your concern is that the invocation log entry will be there without any indication on whether or not the user selected yes or no.  _That's your concern, right?_  If so, then I think the existence (or absence) of the subsequent [`Invoke-GHRestMethod` description log entry](https://github.com/microsoft/PowerShellForGitHub/blob/fcfe0c73b03be8f60ed0c75ced572240ff6769e1/GitHubCore.ps1#L206) can act as the confirmation that the action was confirmed by the user to continue.

_Aside_: You still have `Write-InvocationLog` twice in this function...once at the top and then again at 618.

Reply:
/cc: @X-Guardian - not sure if you had seen this response above earlier.

Reply:
So is the final conclusion. to have the `Write-InvocationLog` at the top of each and every function?

Reply:
That is my suggestion given everything else.  It's also how I proceeded when applying the concepts from this PR to #110 and #172.

Reply:
Ok, I'll bring this PR inline with this then.